### ISSUE TITLE
Colour and text formatting support on mobile phones

### ISSUE THREAD

**Issue by GewoonYorick** [thelounge/thelounge#2918]
Currently we can use text formatting with keyboard shortcuts, CTRL +K for colours, CTRL +B for bold ect. However on Phones (and on most other devices without a physical keyboard) this is not a option.

My suggestion, add a button next to the send and upload buttons to get a popup where you can select the text formatting (just make it insert the text formatting symbols like the keyboard shortcuts do).

Another option could be with autocomplete feature (like with the emoji) however my preference goes to the option with a additional button.

**Comment by McInkay** [comment]
Yeah, this would be a nice addition. I think we'll need to have a quick chat about how we'd want to design it, but I like the idea of a "formatting" button.

**Comment by veryclevermind** [comment]
Mcinkay any updates on this one ? if you are making this for Mobile you should also make this for the Desktop client as well. I see that when we do styling, it does not lock to that color same goes for the Bold/italic option. I want to use this Lounge on my https://www.gupshupcorner.com/ and there is like a whole lot of traffic there. It will be great for theLounge so you guys can check how does lounge work on a site where there is a lot of traffic.

**Comment by veryclevermind** [comment]
@McInkay

**Comment by McInkay** [comment]
@veryclevermind I wasn't saying I would work on it, just that we needed to do that if someone wanted to implement this. It's not high on the priority list.

**Comment by veryclevermind** [comment]
@McInkay Any plans on doing this work? I want to implement thelounge on my website, I am unable to do this if there is no option in the GUI to change color or to make font Bold and Italic. If there is a possibility of having a button for Emos too it will be great too. I have been using mibbit from a long time because of its user-friendly GUI( Also it has the remove chat feature: where people can remove chat ). I am sure if i use theLounge on Gupshup it will be great for the Lounge too there is a traffic of more than 22k daily.

**Comment by McInkay** [comment]
No, as mentioned, it's not high on the priority list.

It would be a nice thing to have, but I do not have the time to be doing it as I have a lot of other things to do. Sorry, we only do this in our spare time, and if you want to have more support for these things, then feel free to pay us to work full time on it :-P 

> I am sure if i use theLounge on Gupshup it will be great for the Lounge too there is a traffic of more than 22k daily

Yeaaaaah, that's not going to convince anyone.

We have plenty of users, the issue is people to contribute code/money just now. More users = more problems.

**Comment by veryclevermind** [comment]
ready to pay for this option any idea for an estimate? a private conversation would be much better?

**Comment by dgw** [comment]
@veryclevermind At least take the conversation to #thelounge on freenode, rather than here, probably.

**Comment by richrd** [comment]
I was messing with this as I was bored. I made a prototype that opens a formatting toolbar when you select text. See proof of concept video: https://lounge.bittemple.org/uploads/2542de531b00f58a/SVID_20200105_211943_1.mp4

This definitely needs discussion on how it should work. After playing with it I noticed that detecting when the user selects text is a bit unreliable and a formatting button might be better.

**Comment by richrd** [comment]
I ditched showing the toolbar when selecting text in the input field and replaced it with a toggle button next to the upload button. Prototype can be tested in this branch if anyone wants to take a look https://github.com/thelounge/thelounge/tree/richrd/formatting-palette
